plugins/identity-shiro/src/main/java/org/opensearch/identity/shiro/ShiroTokenManager.java

```diff
@@ -51,15 +53,14 @@ public Optional<AuthenticationToken> translateAuthToken(org.opensearch.identity.
             final BasicAuthToken basicAuthToken = (BasicAuthToken) authenticationToken;
             return Optional.of(new UsernamePasswordToken(basicAuthToken.getUser(), basicAuthToken.getPassword()));
         }
-
         return Optional.empty();
     }
 
     @Override
-    public AuthToken issueToken(String audience) {
+    public AuthToken issueOnBehalfOfToken(List<String> claims) {
```

Comment:
Can this be a map of claims?

i.e. the request can pass a map like:

```
{
   "aud": "ext1",
   ...
}
```

it would be up to the implementer to use that map of claims and some should be forbidden from being passed in, but determined in the implementation of `issueOnBehalfOfToken`. For example, `sub` - the username of the currently authenticated user, `exp` - the expiry and `iat` - issued at should be forbidden from the caller of this method to include in the payload of the token and should be determined within the implementation of `issueOnBehalfOfToken` other claims may be permittable or it could be an allowlist style where `aud` is allowed to be passed in.

Reply:
Sure, I can swap to a map. I am not sure what all of the options will be though so we will probably have to change things in the future.

Reply:
Generic String to Object (String or primitive) may be fine for first implementation and left up to the implementer to determine what to do with what was passed in. If we needed to make it more restrictive with what keys were passed in then it could come from an enum, but I think it makes sense to keep it generic and left up to the implementer how to handle what is passed in.

Reply:
Yeah sounds good to me.